### ISSUE TITLE
render_template broken with AppDaemon 4.4

### ISSUE THREAD

**Issue by droans** [AppDaemon/appdaemon#1747]
### What happened?
`render_template` is no longer working in the latest version. 

### Version

4.4.2

### Installation type

Docker container

### Relevant log output

```sh
ERROR HASS: Unexpected error during call_plugin_service()
ERROR HASS: Service: default.template.render Arguments: {'template': "{{ states('sun.sun') }}"}
ERROR HASS: ------------------------------------------------------------
ERROR HASS: Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/appdaemon/plugins/hass/hassplugin.py", line 556, in call_plugin_service
    r = await self.session.post(api_url, json=data)
  File "/usr/local/lib/python3.10/site-packages/aiohttp/client.py", line 508, in _request
    req = self._request_class(
  File "/usr/local/lib/python3.10/site-packages/aiohttp/client_reqrep.py", line 305, in __init__
    self.update_host(url)
  File "/usr/local/lib/python3.10/site-packages/aiohttp/client_reqrep.py", line 364, in update_host
    raise InvalidURL(url)
aiohttp.client_exceptions.InvalidURL: /api/template
```

### Relevant code in the app or config file that caused the issue

```sh
class TestApp(AppBase):
    def initialize(self):

        template = "{{ states('sun.sun') }}"
        x = self.render_template(template)
        self.log(x)
```

**Comment by joBr99** [comment]
just wanted to open the exact same issue :) with the 0.12.0 version of the HA Addon this now also hit HA OS Users

**Comment by joBr99** [comment]
not sure about this, try to restore the backup

**Comment by acockburn** [comment]
Thanks for reporting, I’ll take a look.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by Justihar** [comment]
I created a PR that hopefully resolves this issue.

**Comment by jrspowers** [comment]
> I created a PR that hopefully resolves this issue.

when will this come out or can it already be installed on home assistant?

**Comment by Justihar** [comment]
It is currently available in dev but I don't know when a release will be pushed out.

**Comment by jrspowers** [comment]
> It is currently available in dev but I don't know when a release will be pushed out.

can't go into dev mode ?  or is that not possible?

**Comment by joBr99** [comment]
Just recert to the previous version ...

**Comment by jrspowers** [comment]
> Just revert to the previous version ...

If only it were that easy it's not like choosing the previous version reinstalling just like on hacs this one has to be backed up that won't work for me I've been messing around today unfortunately otherwise I've done it a long time ago

**Comment by brett-todd** [comment]
Thanks @joBr99, restoring previous AppDaemon backup has worked and nspanel-lovelace output now displays as expected.

**Comment by jrspowers** [comment]
> Thanks @joBr99, restoring previous AppDaemon backup has worked and nspanel-lovelace output now displays as expected.

yes I wanted to do that too but my backup says very long backup does not exist very unfortunate so I can't go back so wait for a new update

**Comment by brett-todd** [comment]
@jrspowers You might be able to workaround by editing your local version of hassplugin.py and restarting the daemon with the following fix merged to dev:

![image](https://user-images.githubusercontent.com/39944380/231899801-cd2bf06c-69e5-4d3f-aa76-f88a161924d2.png)

**Comment by jrspowers** [comment]
> @jrspowers You might be able to workaround by editing your local version of hassplugin.py and restarting the daemon with the following fix merged to dev:
> 
> 
> 
> ![image](https://user-images.githubusercontent.com/39944380/231899801-cd2bf06c-69e5-4d3f-aa76-f88a161924d2.png)
> 
> 

hello so if i understand correctly i need to put what in green & remove red & then restart appdaemon & then what about dev mode ?  thanks

**Comment by xiconfjs** [comment]
> hello so if i understand correctly i need to put what in green & remove red & then restart appdaemon & then what about dev mode ? thanks

Hi, I'm in the same situation so here is my hotfix:

1. login to your home assistant vm
`ssh -p22222 root@<IP>`
2. After login grep the id of the appdaemon docker instance
`docker ps | grep appdaemon | awk '{ print $1}'`
3. Use the id to start a shell into this appdaemon docker instance
`docker exec -ti <docker-id> bash`
4. Inside the docker instance find the relevant file
`find / -iname "hassplugin.py"`
5. Edit the file with **vi** as shown here [#1750 commit](https://github.com/AppDaemon/appdaemon/pull/1750/commits/f9f5b9295ecbbbfe60e12c258280b4e5ecb1dc32)
6. After saving the find the appdaemon process id and kill it
`kill $(ps aux | grep /usr/bin/appdaemon | grep -v grep | awk '{ print $1 }')`
7. Now the fixed version of appdaemon should be active and you can leave the docker instance via **Ctrl-D**

**Comment by brett-todd** [comment]
Yes, don't worry about "dev" comment as just referring to this fix being pushed into the AppDaemon dev branch. This manual local adjustment will get overwritten when next version of AppDaemon is released

**Comment by jrspowers** [comment]
> Hi, I'm in the same situation so here is my hotfix:
> 
> login to your home assistant vm
> ssh -p22222 root@<IP>
> After login grep the id of the appdaemon docker instance
> docker ps | grep appdaemon | awk '{ print $1}'
> Use the id to start a shell into this appdaemon docker instance
> docker exec -ti <docker-id> bash
> Inside the docker instance find the relevant file
> find / -iname "hassplugin.py"
> Edit the file with vi as shown here #1750 commit
> After saving the find the appdaemon process id and kill it
> kill $(ps aux | grep /usr/bin/appdaemon | grep -v grep | awk '{ print $1 }')
> Now the fixed version of appdaemon should be active and you can leave the docker instance via Ctrl-D

everything first thanks for this nice explanation but I have a few questions 
1 I don't have it running in docker but in ha itself 
2 who log in to ssh via login & password that I always use to mean ssh or the ones you put here?  
3 where can I find that ID?  thanks in advance!

**Comment by xiconfjs** [comment]
> everything first thanks for this nice explanation but I have a few questions
> 1 I don't have it running in docker but in ha itself

perhaps I'm mistaken, but usually all addons are running in separated docker containers with running inside the ha vm

> 2 who log in to ssh via login & password that I always use to mean ssh or the ones you put here?

for accessing the home assistant vm via ssh (and not the ssh addon container) follow [this guide](https://developers.home-assistant.io/docs/operating-system/debugging/)

> 3 where can I find that ID? thanks in advance!

the "docker-id" can be found with the command I posted before and here again:
`docker ps | grep appdaemon | awk '{ print $1}'`

If you have further questions, it would be better to discuss this outside of this issue.

**Comment by jrspowers** [comment]
> > everything first thanks for this nice explanation but I have a few questions
> 
> > 1 I don't have it running in docker but in ha itself
> 
> 
> 
> perhaps I'm mistaken, but usually all addons are running in separated docker containers with running inside the ha vm
> 
> 
> 
> > 2 who log in to ssh via login & password that I always use to mean ssh or the ones you put here?
> 
> 
> 
> for accessing the home assistant vm via ssh (and not the ssh addon container) follow [this guide](https://developers.home-assistant.io/docs/operating-system/debugging/)
> 
> 
> 
> > 3 where can I find that ID? thanks in advance!
> 
> 
> 
> the "docker-id" can be found with the command I posted before and here again:
> 
> `docker ps | grep appdaemon | awk '{ print $1}'`
> 
> 
> 
> If you have further questions, it would be better to discuss this outside of this issue.

okay i understand the benefit how can i contact you i think discord ?  my name jarne#4016

**Comment by masterfish1** [comment]
> 1 I don't have it running in docker but in ha itself

I think you are in the same boat as I am. My understanding is that if you are using HAOS, you can't access the individual addon docker containers.

**Comment by xiconfjs** [comment]
> I think you are in the same boat as I am. My understanding is that if you are using HAOS, you can't access the individual addon docker containers.

this could help: https://community.home-assistant.io/t/add-on-hassos-ssh-port-22222-configurator/264109

**Comment by masterfish1** [comment]
> this could help:

That did indeed help (after a bit of Googling on how to use vi and a large amount of PEBCAK). However, after setting up that addon, I wonder if it is needed as I still had to turn off Protection Mode on the HA SSH plugin. Perhaps turning off Protection Mode on its own would be sufficient.

Anyway, many thanks for offering a solution.

**Comment by javamahn** [comment]
Still frustrated even with the addon I cannot get ssh host access to edit the docker file. I like the ease of installation with HASSOS but the lockout of utilities for those who can use vi is ridiculous. Feels more like Apple vs Android. Any timeline on the AppDaemon patch to fix this bug so my NSPanels start working again?

**Comment by joBr99** [comment]
check your backups, if you had the old addon installed you most likely also have a backup of the old addon that you can restore

editing the dockerfile is most likely not what you want to do HA OS isn't using the appdaemon container directly, it's the HA Appdaemon Addon and the addon only has the buggy release bundled, so there is nothing you can do in the dockerfile to fix this

however if you find the hassplugin.py, you should be able to fix it

**Comment by javamahn** [comment]
> check your backups, if you had the old addon installed you most likely also have a backup of the old addon that you can restore
> 
> editing the dockerfile is most likely not what you want to do HA OS isn't using the appdaemon container directly, it's the HA Appdaemon Addon and the addon only has the buggy release bundled, so there is nothing you can do in the dockerfile to fix this
> 
> however if you find the hassplugin.py, you should be able to fix it

Thankfully just as you were replying I found that I did indeed backup 0.11.0 and just restored it and I am back up again. I will be sure to backup all addon upgrades now. It does make me question the limitations on being able to fix issues myself. Thank joBr99 for your efforts.

**Comment by joBr99** [comment]
posted steps to fix the issue with 0.12.0 of the addon in the issue over here:
https://github.com/joBr99/nspanel-lovelace-ui/issues/847#issuecomment-1509878292

you don't have to use sed, there is also vi installed in the docker container ;)

**Comment by rossi75** [comment]
> > check your backups, if you had the old addon installed you most likely also have a backup of the old addon that you can restore
> > editing the dockerfile is most likely not what you want to do HA OS isn't using the appdaemon container directly, it's the HA Appdaemon Addon and the addon only has the buggy release bundled, so there is nothing you can do in the dockerfile to fix this
> > however if you find the hassplugin.py, you should be able to fix it
> 
> Thankfully just as you were replying I found that I did indeed backup 0.11.0 and just restored it and I am back up again. I will be sure to backup all addon upgrades now. It does make me question the limitations on being able to fix issues myself. Thank joBr99 for your efforts.

Well, i am running ha OS on a raspberry 3 B+. Before the Installation of Version 0.12.0, haos did a Backup of 0.11.0 Version. 
I switched offen the automatic update, hence tried to revert to 0.11.0. The first dialogue does not close after choosing the appdaemon to revert back. 

Any hints for this? 

The other workaround does not work for me as the Linux cannot find the hassplugin.py......

**Comment by joBr99** [comment]
> The other workaround does not work for me as the Linux cannot find the hassplugin.py......

Did you do this or the stuff on the terminal mentioned ealier in my comment?


> Configure the following command in the appdaemon addon settings:
>
>`sed -i 's:"/api/template":f"{self.ha_url}/api/template":g' /usr/lib/python3.10/site-packages/appdaemon/plugins/hass/hassplugin.py`

![image](https://user-images.githubusercontent.com/29555657/232238594-ce054d10-f879-4188-bdba-a849f2a265fc.png)

**Comment by acockburn** [comment]
This will be fixed in 4.4.2 which will be released later today.
